samples/ts-valueentity-shopping-cart/integration-test/shoppingcart.integration-test.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IntegrationTestkit } from "@lightbend/akkaserverless-javascript-sdk";
+import { expect } from "chai"
+import * as proto from "../lib/generated/proto";
+import shoppingcartEnity from '../src/shoppingcart'
+
+type ShoppingCartService = proto.com.example.shoppingcart.ShoppingCartService
+
+type AsyncShoppingCartService = {
+  [K in keyof ShoppingCartService as `${K}Async`]: ShoppingCartService[K]
+}
```

Comment:
@pvlugter this is a pretty cool TS feature, we could move this small piece of type inference logic into the `IntegrationTestkit`, when we are working on the https://github.com/lightbend/akkaserverless-javascript-sdk/issues/168 issue. What do you think?

Reply:
👍🏼 sounds good